### PR TITLE
Adds feature to pass on errors from functions

### PR DESCRIPTION
**Pull request by palnabarun**
Solves #52

### CODE REVIEW

**Comment by palnabarun** [comment]
I wanted to do something like `except Exception` but I couldn't find a method to know the exact error name from that `Exception` object. Have any ideas?

**Comment by palnabarun** [comment]
Can it merged now? @anandology

File: firefly/client.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def handle_response(self, response):
         elif response.status_code == 422:
             raise ValidationError(response.json()["error"])
         elif response.status_code == 500:
-            raise FireflyError("Internal Server Error")
+            raise FireflyError(response.json()["error"])
```
Comment (anandology) [inline firefly/client.py]: What if `response.json()` fails?

You are trying to given valid JSON response on errors, but they can be internal server errors that are not raised by you. You should be able to handle them too. 

Here is how I suggest you to handle the error.

Try to do `.json()` only when the content-type is set to `application/json` and if that fails take the entire string as error message.

Comment (palnabarun) [inline firefly/client.py]: Sure. 
That slipped by me.

**Pull request merged**